frontend/src/utils/customHooks/usePosts.js

```diff
@@ -0,0 +1,22 @@
+import { useSelector } from 'react-redux'
+
+const ALL_POSTS = 'post'
+const CATEGORY_POSTS = 'category'
+const SEARCHED_POSTS = 'searched'
+
+export const usePosts = (type = 'post') => {
```

Comment:
This is going to cause unnecessary renders of the component. 

Let's say you use only `searchedPosts` in your component. When `categoryPosts` changes, your component will render even though it doesn't use it. IMHO better solution is to use one hook per one selector (of course don't forget to extract selector to separate function)